modules/core/src/smithy4s/Nullable.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+import smithy4s.Nullable._
+
+/**
+ * ADT isomorphic to `Option`, but representing types that were passed explicitly as
+ * null rather than ones that were absent.
+ *
+ * The goal of this datatype is to offer the ability to distinguish, during
+ * serialisation, between the absence of a field and the nullity of a field.
+ */
+sealed trait Nullable[+A] {
+  def map[B](f: A => B): Nullable[B] = this match {
+    case Value(a) => Value(f(a))
+    case Null     => Null
+  }
+
+  def fold[B](whenValue: A => B, whenNull: => B): B =
```

Comment:
I think it'd be good to match the signature of Option's fold : 

```
def fold[B](whenNull : => B)(whenValue: A => B) : B
```

Reply:
Shall do!

Also, just realising - would it make sense to put some type classes for `Nullable` in the `cats` module?

Reply:
I suppose it would 👍